docs/bbe-guidelines/bbe-guidelines.md

```diff
@@ -0,0 +1,174 @@
+
+1. The maximum character count per line should be 80 in .bal files. Otherwise, the lines will be wrapped in the [Ballerina](https://ballerina.io/) website reducing the readability.
+
+Example:
+
+
+
+
+<p id="gdcalert1" ><span style="color: red; font-weight: bold">>>>>>  gd2md-html alert: inline image link here (to images/Guidelines-in0.png). Store image on your image server and adjust path/filename if necessary. </span><br>(<a href="#">Back to top</a>)(<a href="#gdcalert2">Next alert</a>)<br><span style="color: red; font-weight: bold">>>>>> </span></p>
+
+
+![alt_text](images/Guidelines-in0.png "image_tooltip")
+
+
+
+
+*   **Add comments to the code blocks as much as possible with “//”. They will move to the RHS box in b.io. So comments are used as a mechanism to describe the code.**
+
+Ex:
+
+Code
+
+
+
+<p id="gdcalert2" ><span style="color: red; font-weight: bold">>>>>>  gd2md-html alert: inline image link here (to images/Guidelines-in1.png). Store image on your image server and adjust path/filename if necessary. </span><br>(<a href="#">Back to top</a>)(<a href="#gdcalert3">Next alert</a>)<br><span style="color: red; font-weight: bold">>>>>> </span></p>
+
+
+![alt_text](images/Guidelines-in1.png "image_tooltip")
+
+
+b.io:
+
+
+
+<p id="gdcalert3" ><span style="color: red; font-weight: bold">>>>>>  gd2md-html alert: inline image link here (to images/Guidelines-in2.png). Store image on your image server and adjust path/filename if necessary. </span><br>(<a href="#">Back to top</a>)(<a href="#gdcalert4">Next alert</a>)<br><span style="color: red; font-weight: bold">>>>>> </span></p>
+
+
+![alt_text](images/Guidelines-in2.png "image_tooltip")
+
+
+
+
+*   **Since comments are used in RHS they should be a valid sentence. (i.e. Start with upper case letter, ends with full stop etc)**
+*   **A comment applies to the subsequent lines in the file until another comment or an empty line is found. Use comments/new lines appropriately to ensure that it applies only to the relevant lines.**
+*   **No restriction for the max chars in comment lines as they will go to RHS side and wraps automatically. But since users can refer the code in GitHub, it is better if we can have the same char limit as the code line (i.e. 80) as it increases the readability of the code file.**
+
+Ex: Comments are significantly longer than the code line, which is not readable. 
+
+
+
+<p id="gdcalert4" ><span style="color: red; font-weight: bold">>>>>>  gd2md-html alert: inline image link here (to images/Guidelines-in3.png). Store image on your image server and adjust path/filename if necessary. </span><br>(<a href="#">Back to top</a>)(<a href="#gdcalert5">Next alert</a>)<br><span style="color: red; font-weight: bold">>>>>> </span></p>
+
+
+![alt_text](images/Guidelines-in3.png "image_tooltip")
+
+
+
+
+*   **All keywords and any other word which needs to be highlighted should be used with backquotes. (ex:  `xml`).  Don’t use a single quote as it won’t get highlighted. **
```

Comment:
Formatting words/phrases as code should be done taking the context in to account. We should not be blindly formatting for the sake of adhering to this guideline. For example take the word "record" or "object". These are keywords in the language. But whether to format them as code or not depends on the context. If you are simply referring to the record construct in the language, it should not be highlighted. For example consider the below excerpt from the records BBE. 

> // Defines an open record type named `Student`. The `{` and `}` delimiters indicate that in addition to the defined fields,
// this record type allows additional fields with pure-typed (i.e., `anydata|error`) values.
// The descriptor `record { }` is equivalent to `record {| (anydata|error)...; |}`.
type Student record {
    string name;
    int age;
    Grades grades;
};

Note how record is not highlighted when the context in which it appears is not code.

Reply:
Added the above as a note.